src/modules/cpu/kernel/resize.hpp

```diff
@@ -0,0 +1,1109 @@
+        Rpp32f wOffset = (wRatio - 1) * 0.5f;
+        Rpp32s kernelSize = 2;
+        Rpp32s noOfCoeffs = 4; // kernelSize * kernelSize
+        Rpp32s vectorIncrementPerChannel = 8;
+        Rpp32s vectorIncrementPkd = 24;
+
+        Rpp8u *srcPtrChannel, *dstPtrChannel, *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+        srcPtrChannel = srcPtrImage + (roiPtr->xywhROI.xy.y * srcDescPtr->strides.hStride) + (roiPtr->xywhROI.xy.x * srcLayoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        Rpp32u alignedLength = dstImgSize[batchCount].width & ~7;
+        __m256 pWRatio = _mm256_set1_ps(wRatio);
+        __m256 pWOffset = _mm256_set1_ps(wOffset);
+        __m256 pWidthLimit = _mm256_set1_ps((float)widthLimit);
+        __m256 pDstLocInit = _mm256_setr_ps(0, 1, 2, 3, 4, 5, 6, 7);
+        __m256 pWeightParams[noOfCoeffs], pBilinearCoeffs[noOfCoeffs], pDstLoc;
+        Rpp32f weightParams[noOfCoeffs], bilinearCoeffs[noOfCoeffs];
+        Rpp32s srcLocationColumnArray[8] = {0};     // Since 8 dst pixels are processed per iteration
+        Rpp32s srcLocationRow, srcLocationColumn;
+
+        // Resize with fused output-layout toggle (NHWC -> NCHW)
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            Rpp8u *dstPtrRowR, *dstPtrRowG, *dstPtrRowB;
+            dstPtrRowR = dstPtrChannel;
+            dstPtrRowG = dstPtrRowR + dstDescPtr->strides.cStride;
+            dstPtrRowB = dstPtrRowG + dstDescPtr->strides.cStride;
+
+            for(int dstLocRow = 0; dstLocRow < dstImgSize[batchCount].height; dstLocRow++)
+            {
+                Rpp8u *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+
+                Rpp8u *srcRowPtrsForInterp[2];
+                compute_resize_src_loc(dstLocRow, hRatio, heightLimit, srcLocationRow, &weightParams[0], hOffset);
+                srcRowPtrsForInterp[0] = srcPtrChannel + srcLocationRow * srcDescPtr->strides.hStride;    // srcPtrTopRow for bilinear interpolation
+                srcRowPtrsForInterp[1]  = srcRowPtrsForInterp[0] + srcDescPtr->strides.hStride;           // srcPtrBottomRow for bilinear interpolation
+                pWeightParams[0] = _mm256_set1_ps(weightParams[0]);
+                pWeightParams[1]  = _mm256_set1_ps(weightParams[1]);
+                pDstLoc = pDstLocInit;
+
+                int vectorLoopCount = 0;
+                for (; vectorLoopCount < alignedLength; vectorLoopCount += vectorIncrementPerChannel)
+                {
+                    __m256 pSrc[12], pDst[3];
+                    compute_resize_src_loc_avx(pDstLoc, pWRatio, pWidthLimit, srcLocationColumnArray, &pWeightParams[2], pWOffset, true);
```

Comment:
For sections like these from L108-L112, you may want to leave a brief comment on the right for better readability - saying what each helper does

Reply:
I have added the comments please check